paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -196,19 +184,55 @@ internal class DefaultPaymentSheetLoader @Inject constructor(
             )
         }
 
-        val sortedPaymentMethods = async {
-            paymentMethods.await().withLastUsedPaymentMethodFirst(savedSelection.await())
+        val customer = async {
+            when (customerConfig?.accessType) {
+                is PaymentSheet.CustomerAccessType.CustomerSession -> {
+                    elementsSession.customer?.let { customer ->
+                        CustomerState(
+                            id = customer.session.customerId,
+                            ephemeralKeySecret = customer.session.apiKey,
+                            paymentMethods = customer.paymentMethods
+                                .withLastUsedPaymentMethodFirst(savedSelection.await()),
+                        )
+                    } ?: run {
+                        val exception = IllegalStateException(
+                            "Excepted 'customer' attribute as part of 'elements_session' response!"
+                        )
+
+                        errorReporter.report(
+                            ErrorReporter
+                                .UnexpectedErrorEvent
+                                .PAYMENT_SHEET_LOADER_ELEMENTS_SESSION_CUSTOMER_NOT_FOUND,
+                            StripeException.create(exception)
+                        )
+
+                        throw exception
```

Comment:
We shouldn't expect `customer` field in `ElementsSession` to be `null` when we are using `CustomerSession`. I'm unsure however if we should entirely fail the loading process here or just continue with a no customer payment session.

Reply:
I think it's fine to fail the load. But we should probably do what we do for the other API: https://github.com/stripe/stripe-android/blob/89b8761a953c1687d21a769f3f9d718c03df087e/paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt#L254

That being said, we will get double reporting of this error, as is. Which I think is also what we're doing in the other implementation, but we have more analytics for that one (that it worked correctly too).